### PR TITLE
Fixing Bittensor and Hivemind Async Issues

### PR DESCRIPTION
**Pull request by KMFODA**
Converting to continuous training using HF's trainer and removing BlockingIO errors by changing the async forward calls on the miner + validator side.

### CODE REVIEW

File: neurons/validator.py [hunk 1]
```diff
@@ -93,9 +127,17 @@ async def forward(self):
...
+        # The validator is now initialized and ready to use
+        await validator.run()
+
 if __name__ == "__main__":
-    with Validator() as validator:
-        while True:
-            # bt.logging.info("Validator running...", time.time())
-            time.sleep(5)
\ No newline at end of file
+    asyncio.run(main())
```
Comment (mikkelmedm) [inline neurons/validator.py:137]: Can go back to the old way of starting the Validator, if the blockingIOError is fixed with using a dict instead. asyncio.run() is used here to try to remedy that error, so probably better to revert to the old method of starting the validator. Should also be changed inside template/base/validator.

File: template/base/validator.py
Comment (mikkelmedm) [inline template/base/validator.py:1]: So probably all the changes here should just be reverted to the previous commit - after removing asyncio + async function calls from the neurons/validator.py

File: template/validator/reward.py
Comment (mikkelmedm) [inline template/validator/reward.py:1]: Same as here. The await calls and the async function defitions should be reverted

File: template/validator/validator_core.py
Comment (mikkelmedm) [inline template/validator/validator_core.py:1]: Async and await calls here as well

File: neurons/miner.py
Comment (mikkelmedm) [inline neurons/miner.py:1]: Not sure if using global variables is the best way to communicate between the CustomValidationCallback and the Trainer.

**Review by mikkelmedm — APPROVED** [review]
Some changes irt. to not using the asyncio setup for the validator + miner changes

**Comment by KMFODA** [comment]
Closing this as most of these changes have been merged through other branches.

**Pull request closed without merging**